### PR TITLE
Make backwards-compatible changes for Xcode 7

### PR DESCRIPTION
**Pull request by tgoyne**
This makes all of the obj-c stuff build and pass tests in both Xcode 6 and 7. Swift stuff obviously doesn't build in Xcode 7, but this includes all of the changes needed for it that don't break 6 to cut down on the diff a bit.

### CODE REVIEW

File: Realm/Tests/RLMMultiProcessTestCase.m [hunk 1]
```diff
@@ -77,6 +75,7 @@ - (NSTask *)childTask {
     NSString *testName = [NSString stringWithFormat:@"%@/%@", self.className, self.testName];
     NSMutableDictionary *env = [NSProcessInfo.processInfo.environment mutableCopy];
     env[@"RLMProcessIsChild"] = @"true";
+    [env removeObjectForKey:@"XCTestConfigurationFilePath"];
```
Comment (bdash) [inline Realm/Tests/RLMMultiProcessTestCase.m]: Why is this necessary?

Comment (tgoyne) [inline Realm/Tests/RLMMultiProcessTestCase.m]: Xcode 7 switched from passing all the undocumented options for things like how to communicate with the IDE from the command line to a config file, with the path specific in the env variable. We don't want the child process to use these settings because multiple xctest processes communicating with the IDE doesn't work at all.

File: Realm/Swift/RLMSupport.swift [hunk 2]
```diff
@@ -29,18 +29,29 @@ extension RLMObject {
...
-        var i: UInt  = 0
-
-        return GeneratorOf<RLMObject> {
-            if (i >= self.count) {
-                return .None
-            } else {
-                return self[i++] as? RLMObject
-            }
-        }
+        return GeneratorOf(RLMGenerator(collection: self))
```
Comment (bdash) [inline Realm/Swift/RLMSupport.swift]: Why is this necessary?

File: Realm/Swift/RLMSupport.swift [hunk 3]
```diff
@@ -29,18 +29,29 @@ extension RLMObject {
...
 
+public class RLMGenerator: GeneratorType {
+    private var i: UInt = 0
+    private let collection: RLMCollection
+
+    init(collection: RLMCollection) {
+        self.collection = collection
+    }
+
+    typealias Element = RLMObject
```
Comment (segiddins) [inline Realm/Swift/RLMSupport.swift]: I think this can be inferred?

File: RealmSwift/Tests/ObjectCreationTests.swift [hunk 4]
```diff
@@ -441,7 +441,7 @@ class ObjectCreationTests: TestCase {
         Realm().commitWrite()
         switch type {
             case .Bool:     return ["invalid", 2 as Int, 1.1 as Float, 11.1 as Double]
-            case .Int:      return ["invalid", true, false, 1.1 as Float, 11.1 as Double]
```
Comment (segiddins) [inline RealmSwift/Tests/ObjectCreationTests.swift]: why this change?

File: Realm/Swift/RLMSupport.swift (hunk 2 above)
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: It allows reusing RLMGenerator for both RLMArray and RLMResults, I guess.

Comment (segiddins) [inline Realm/Swift/RLMSupport.swift]: But why not just return `RLMGenerator`?

File: RealmSwift/Tests/ObjectCreationTests.swift [hunk 5]
```diff
@@ -441,7 +441,7 @@ class ObjectCreationTests: TestCase {
...
+            case .Int:      return ["invalid", 1.1 as Float, 11.1 as Double]
```
Comment (jpsim) [inline RealmSwift/Tests/ObjectCreationTests.swift]: So because of your changes to `nsnumber_is_like_integer`, Boolean values can now be assigned to `Int` properties? Why?

File: Realm/Swift/RLMSupport.swift (hunk 2 above)
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: Right, and then the `Element` type in `RLMGenerator` should probably be inferred.

Comment (tgoyne) [inline Realm/Swift/RLMSupport.swift]: GeneratorOf went away in Swift 2.0, and there doesn't appear to be any equivalent generator from a block in the standard library, so it needs an explicit generator type (which turns out to be less code anyway...). Returning the type-erased generator isn't really necessary; it's just much more natural in 2.0 because it's a protocol (AnyGenerator) rather than a struct wrapping the inner generator.

File: Realm/Swift/RLMSupport.swift (hunk 3 above)
Comment (tgoyne) [inline Realm/Swift/RLMSupport.swift]: It wasn't.

File: RealmSwift/Tests/ObjectCreationTests.swift (hunk 5 above)
Comment (tgoyne) [inline RealmSwift/Tests/ObjectCreationTests.swift]: In obj-c assigning a boolean value to an int property is perfectly legal and sensible, and NSNumber started actually returning a boolean type from objcType when given boolean boxed in obj-c.

File: Realm/Swift/RLMSupport.swift (hunk 2 above)
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: This works:

```swift
public class RLMGenerator: GeneratorType {
    private var i: UInt = 0
    private let collection: RLMCollection

    init(collection: RLMCollection) {
        self.collection = collection
    }

    public func next() -> RLMObject? {
        if i >= collection.count {
            return .None
        } else {
            return collection[i++] as? RLMObject
        }
    }
}

extension RLMArray: SequenceType {
    // Support Sequence-style enumeration
    public func generate() -> RLMGenerator {
        return RLMGenerator(collection: self)
    }
}
```

**Comment by segiddins** [comment]
This looks reasonably sane

**Comment by segiddins** [comment]
:+1:

File: RealmSwift/Tests/ObjectCreationTests.swift (hunk 4 above)
Comment (jpsim) [inline RealmSwift/Tests/ObjectCreationTests.swift]: Shouldn't we now add `true` and `false` to `validValuesForSwiftObjectType` for Int?

File: RealmSwift/Tests/PerformanceTests.swift [hunk 6]
```diff
@@ -19,8 +19,6 @@
 import XCTest
 import RealmSwift
 
-#if !DEBUG && os(iOS)
```
Comment (jpsim) [inline RealmSwift/Tests/PerformanceTests.swift]: :tada:

**Comment by jpsim** [comment]
:+1:
